src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -1297,6 +1298,70 @@ void SolverInterfaceImpl::readScalarData(
   PRECICE_DEBUG("Read value = {}", value);
 }
 
+void SolverInterfaceImpl::setBoundingBoxes(
+    const int     meshID,
+    const double *boundingBox,
+    const int     size) const
+{
+  PRECICE_TRACE(meshID);
+  PRECICE_REQUIRE_MESH_USE(meshID);
+  if (size == 0)
+    return;
+  PRECICE_CHECK(boundingBox != nullptr, "The provided bounding boxes are empty.");
+  //  PRECICE_REQUIRE_MESH_MODIFY(meshID);
```

Comment:
What are the requirements in this case?
It would be good to define them here and come up with a similar macro.

Reply:
The requirements are calling it before `initialize`, calling it only once and the mesh needs to be used. All conditions are checked now, I hope that's sufficient.